atlp-rwanda/atlp-pulse-fn
# ft: add and update ratings

# PR Description

We have to make sure that the coordinator can add a review and can update it.

# Description of tasks that were expected to be completed

- [x] coordinator add ratings to trainees
- [x] coordinator add feedback on rate
- [x] trainee add feedback on rate
- [x] update ratings
- [x] admin approve ratings updates

# How has this been tested?
* For unity nesting, run `npm test` command to test
* For verifying the functionality
   ```bash
    git clone https://github.com/atlp-rwanda/atlp-pulse-fn
    cd atlp-pulse-fn
    ```
    go to the branch called `ft-add-ratings` then
    ```bash
     npm install
     npm run dev
     open http://localhost:3000
    ```
    - ### for adding ratings
        - login as a coordinator with these credentials:  
             `email: georgia.gonzales@devpulse.com`
              `Password : Test@12345`
        - in side bar, click on Ratings link to reach ratings
        - click on `add new ratings` button
        - add required data and continue
    - ### adding feedback to rating remarks
        - login as a coordinator
        - 
           * for `coordinator` go to ratings in sidebar, you will find `chat icon` button in the ratings table
           * for `trainee` go to performance in sidebar, i you will find `details` button in the ratings table
        - then click it to open model
        - the add your feedback
    - ### Updating rating
        - login as a coordinator
        - go to ratings in sidebar, you will find `write icon` button in the ratings table
        - then click it to open model
        - the add your updates then submit
    - ### approve and reject updates
        - login as a admin
        - go to `Updated Ratings` in sidebar
        - You will find 2 buttons in the table
          * click ✅ button approve
          * click ❌ to reject 
        - those button will open models and click the `approve` or `reject` buttons to continue
# Number of Commits

**1 commit**

# Screenshots (If appropriate)
<img width="1440" alt="Screenshot 2023-06-30 at 11 01 59" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/66f4d8c9-f5d1-4c6a-992b-e433fa12ca40">
<img width="986" alt="Screenshot 2023-06-30 at 11 02 54" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/3474cedc-c3dd-44a0-b07e-efe62b447500">
<img width="880" alt="Screenshot 2023-06-30 at 11 17 38" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/a5cacb1a-8ed0-472c-9ef0-c6888946ac23">
<img width="1440" alt="Screenshot 2023-07-12 at 17 23 53" src="https://github.com/atlp-rwanda/atlp-pulse-fn/assets/112763782/aa4305b1-a640-4367-873c-796a1907341e">


# Please check this Checklist before you submit your PR:
